google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryWatchTest.java

```diff
@@ -441,6 +444,69 @@ public void limitToLast() throws Exception {
     listenerAssertions.addedIdsIsAnyOf(emptyList(), asList("doc2", "doc3"));
   }
 
+  /** Verifies that the SDK orders vector fields the same way as the backend. */
+  @Test
+  public void vectorFieldOrder() throws Exception {
+    // We validate that the SDK orders the vector field the same way as the backend
+    // by comparing the sort order of vector fields from a Query.get() and
+    // Query.addSnapshotListener(). Query.addSnapshotListener() will return sort order
+    // of the SDK, and Query.get() will return sort order of the backend.
+
+    // Test data in the order that we expect the backend to sort it.
+    List<Map<String, Object>> docsInOrder =
+        Arrays.asList(
+            map("embedding", Arrays.asList(1, 2, 3, 4, 5, 6)),
+            map("embedding", Arrays.asList(100)),
+            map("embedding", FieldValue.vector(new double[] {Double.NEGATIVE_INFINITY})),
+            map("embedding", FieldValue.vector(new double[] {-100})),
+            map("embedding", FieldValue.vector(new double[] {100})),
+            map("embedding", FieldValue.vector(new double[] {Double.POSITIVE_INFINITY})),
+            map("embedding", FieldValue.vector(new double[] {1, 2})),
+            map("embedding", FieldValue.vector(new double[] {2, 2})),
+            map("embedding", FieldValue.vector(new double[] {1, 2, 3})),
+            map("embedding", FieldValue.vector(new double[] {1, 2, 3, 4})),
+            map("embedding", FieldValue.vector(new double[] {1, 2, 3, 4, 5})),
+            map("embedding", FieldValue.vector(new double[] {1, 2, 100, 4, 4})),
+            map("embedding", FieldValue.vector(new double[] {100, 2, 3, 4, 5})),
+            map("embedding", map("HELLO", "WORLD")),
+            map("embedding", map("hello", "world")));
```

Comment:
What about the empty map or the empty vector?

Reply:
TY. Added empty map. The backend rejects empty vectors, so I did not add that case.